File: app/src/main/java/com/hop/pirate/HopApplication.java
package com.hop.pirate;

import android.app.Application;
import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.util.Log;

import com.kongzue.dialog.util.DialogSettings;
import com.tencent.bugly.crashreport.CrashReport;

import io.reactivex.functions.Consumer;
import io.reactivex.plugins.RxJavaPlugins;

public class HopApplication extends Application {
    private static final String TAG = "HopApplication";
    private static HopApplication sApplication;
    private boolean isRunning;

    @Override
    public void onCreate() {
        super.onCreate();
        HopApplication.sApplication = this;
        DialogSettings.style = DialogSettings.STYLE.STYLE_IOS;
        DialogSettings.modalDialog = true;
        if (!BuildConfig.DEBUG) {
            CrashReport.initCrashReport(getApplicationContext(), "cfdc2c8a96", false);
        }
        RxJavaPlugins.setErrorHandler(new Consumer<Throwable>() {
            @Override
            public void accept(Throwable throwable) throws Exception {
                Log.d(TAG, "accept: " + throwable.getMessage());
            }
        });

    }

    public static HopApplication getApplication() {
        return HopApplication.sApplication;
    }

    public boolean isRunning() {
        return isRunning;
    }

    public void setRunning(boolean isRunning) {
        this.isRunning = isRunning;
    }

}

File: app/src/main/java/com/hop/pirate/HopApplication.kt
package com.hop.pirate

import android.app.Application
import android.util.Log
import com.kongzue.dialog.util.DialogSettings
import com.tencent.bugly.crashreport.CrashReport
import io.reactivex.plugins.RxJavaPlugins
import kotlin.properties.Delegates

class HopApplication : Application() {
    var isRunning = false
    override fun onCreate() {
        super.onCreate()
        instance = this
        DialogSettings.style = DialogSettings.STYLE.STYLE_IOS
        DialogSettings.modalDialog = true
        if (!BuildConfig.DEBUG) {
            CrashReport.initCrashReport(applicationContext, "cfdc2c8a96", false)
        }
        RxJavaPlugins.setErrorHandler { throwable -> Log.d(TAG, "accept: " + throwable.message) }
    }

    companion object {
        private const val TAG = "HopApplication"
        var instance: HopApplication by Delegates.notNull()
    }
}